### PR TITLE
Introduce Processor Interface

### PR DESCRIPTION
**Pull request by HerbertJordan**
This PR introduces a new interface for a *Transaction Processor*. A transaction processor is tasked in processing incoming transactions by checking the validity of incoming transactions, running them by executing (potentially recursively nested) contract invocations on an interpreter, and performing the necessary world-state modifications upon success.

As part of this, the Interpreter interface `Parameter` type was refined and split into three sub-sections:
 - `BlockParameters` which are parameters that are equivalent for all transactions within a block
 - `TransactionParameters` which are transaction specific parameters
 - `RunContext` which provides access to the world state in the chain (accounts and storage), as well as information of the transaction context (snapshots, log messages, ...);

Additionally, the World State handling API of the `RunContext` has been factored out into a dedicated `WorldState` interface, required in an extended form of the `TransactionContext` as an input parameter for the transaction processing interface.

Furthermore, the structure of the parameters has been simplified. In particular, the `RunContext` no longer includes the function [GetTransactionContext](https://github.com/Fantom-foundation/Tosca/blob/439d3619e83564892d9ee2de9ba21019ca005d82/go/vm/interpreter.go#L73) function. This was a kind of oddity that did not belong the the other functions offered by this interface. Furthermore, all properties retrieved through this function have been exposed at other places in the `Parameters` struct.

This re-organization should simplify the integration of `Processors` and `Interpreters` into host systems aiming to utilize those for block processing.

**Note for Reviewers:** this PR does not alter any features. Preexisting code is expected to work as before. Only some parameters may have been renamed, moved, or consolidated. Some implementations needed to be extended to match the new requirements.

### CODE REVIEW

**Review by facuMH — DISMISSED** [review]
Looks good, nice to have a clean common interface !! just a couple of comments

File: go/geth_adapter/adapter.go [hunk 1]
```diff
@@ -138,19 +138,65 @@ func (a *gethInterpreterAdapter) Run(contract *geth.Contract, input []byte, read
...
+
+	blockParameters := vm.BlockParameters{
+		ChainID:     chainId,
+		BlockNumber: a.evm.Context.BlockNumber.Int64(),
+		Timestamp:   int64(a.evm.Context.Time),
+		Coinbase:    vm.Address(a.evm.Context.Coinbase),
+		GasLimit:    vm.Gas(a.evm.Context.GasLimit),
+		PrevRandao:  difficulty,
+		BaseFee:     baseFee,
+		BlobBaseFee: vm.Value{},
```
[the remark this answers is not in the document]
Comment (HerbertJordan) [reply go/geth_adapter/adapter.go]: Done

File: go/vm/processor.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	// Run executes the transaction provided by the parameters in the specified context.
+	Run(BlockParameters, Transaction, TransactionContext) (Receipt, error)
+}
+
+// Transaction summarizes the parameters of a transaction to be executed on a chain.
+type Transaction struct {
+	Sender     Address       // the sender of the transaction, paying for its execution
+	Recipient  *Address      // the receiver of a transaction, nil if a new contract is to be created
+	Nonce      uint64        // the nonce of the sender account, used to prevent replay attacks
+	Input      []byte        // the input data for the transaction
```
[the remark this answers is not in the document]
Comment (HerbertJordan) [reply go/vm/processor.go]: Done

**Review by facuMH — DISMISSED** [review]

**Review by LuisPH3 — DISMISSED** [review]
This is a little large for me so far, I can only see the surface. 
processor.go is missing license, maybe we should think of a holistic solution for that.

**Review by simonlechner — DISMISSED** [review]
LGTM

**Pull request merged**